src/components/endpoints/ImportProfile.vue

```diff
@@ -0,0 +1,71 @@
+<script setup lang="ts">
+import useWeb3 from '@/compositions/useWeb3'
+import { ref } from 'vue'
+import Notifications from '@/components/Notification.vue'
+import useNotifications from '@/compositions/useNotifications'
+
+const { notification, clearNotification, hasNotification, setNotification } =
+  useNotifications()
+
+const { isAddress } = useWeb3()
+const controllerAddress = ref('')
+
+const onImportProfile = async () => {
+  if (!controllerAddress.value) {
+    return setNotification('Please provide a controller address', 'danger')
+  }
+
+  if (!isAddress(controllerAddress.value)) {
+    return setNotification(
+      'Please provide a valid controller address',
+      'danger'
+    )
+  }
+  try {
+    await window.ethereum.request({
```

Comment:
This call returns the new controller address. We should have a second transaction to give the new controller address permissions on the UP